airflow/providers/apache/hive/hooks/hive.py

```diff
@@ -170,6 +175,10 @@ def _prepare_cli_cmd(self) -> list[Any]:
                 if ";" in proxy_user:
                     raise RuntimeError("The proxy_user should not contain the ';' character")
                 jdbc_url += f";principal={template};{proxy_user}"
+                if self.high_availability:
+                    if proxy_user:
+                        jdbc_url += ";"
```

Comment:
You can check if the jdbc_url ends with ;

Reply:
Yeah makes sense. Let me add that instead of this